TLM/TLM/Util/Record/VehicleRestrictionsLaneRecord.cs

```diff
@@ -53,26 +52,33 @@ public class VehicleRestrictionsLaneRecord : IRecordable {
             }
         }
 
+        /// <summary>
+        /// Obtain vehicle restriction records for applicable lanes within the segment.
+        /// </summary>
+        /// <param name="segmentId">The id of the segment to inspect.</param>
+        /// <returns>
+        /// Returns a list of <see cref="VehicleRestrictionsLaneRecord"/> for all
+        /// restriction-applicable lanes in the segment. The list may be empty if no
+        /// matching lanes were found.
+        /// </returns>
         public static List<VehicleRestrictionsLaneRecord> GetLanes(ushort segmentId) {
-            int maxLaneCount = segmentId.ToSegment().Info.m_lanes.Length;
-            var ret = new List<VehicleRestrictionsLaneRecord>(maxLaneCount);
-            ExtSegmentManager extSegmentManager = ExtSegmentManager.Instance;
-            var lanes = extSegmentManager.GetSortedLanes(
-                segmentId,
-                ref segmentId.ToSegment(),
+
+            var lanes = segmentId.ToSegment().GetSortedLanes(
                 null,
                 LANE_TYPES,
                 VEHICLE_TYPES,
                 sort: false);
+
+            var ret = new List<VehicleRestrictionsLaneRecord>(lanes.Count);
```

Comment:
Can this be zero size, does this happen often? If so, return a premade zero size static readonly list.